core/modules/widgets/view.js

```diff
@@ -30,8 +30,23 @@ ViewWidget.prototype.render = function(parent,nextSibling) {
 	this.parentDomNode = parent;
 	this.computeAttributes();
 	this.execute();
-	if(this.text) {
-		var textNode = this.document.createTextNode(this.text);
+	var textNode;
+	if(this.viewUpdate && this.viewWikified) {
```

Comment:
These flags are a clumsy approach, and will lead to error prone code. If we're really branching the widget into multiple very different implementations then it would be better to encapsulate them as some kind of embedded object. We might call it a "ViewHandler", and it would have render and refresh methods. There would be implementations for plain text and wikified text, with options like plain vs. html that would be passed as constructor flags.

Reply:
Thanks @Jermolene - I'm sure I did it the wrong way around though